api/tracer.go

```diff
@@ -0,0 +1,68 @@
+package api
+
+import (
+	"github.com/alibaba/sentinel-golang/core/base"
+	"github.com/alibaba/sentinel-golang/logging"
+)
+
+var (
+	logger = logging.GetDefaultLogger()
+)
+
+type TracerOptions struct {
+	ctx   *base.EntryContext
+	entry *base.SentinelEntry
+	err   error
+	count int
+}
+
+type TracerOption func(*TracerOptions)
+
+func AddEntry(entry *base.SentinelEntry) TracerOption {
+	return func(opts *TracerOptions) {
+		opts.entry = entry
+	}
+}
+func AddError(err error) TracerOption {
+	return func(opts *TracerOptions) {
+		opts.err = err
+	}
+}
+func AddCount(count int) TracerOption {
+	return func(opts *TracerOptions) {
+		opts.count = count
+	}
+}
+
+func Trace(ctx *base.EntryContext, opts ...TracerOption) {
```

Comment:
Some suggestions:

- The name `Trace` seems too generic. How about `TraceError`?
- Actually it's more common to pass the `Entry` itself instead of the `EntryContext`, so maybe we could provide two functions here: `TraceError(entry, err, opts...)` and `TraceErrorToCtx(ctx, err, opts...)`. Note that the `err` is required (though not used yet).

Reply:
That's great.
I would polish related code.